modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/DataSet.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.data.set;
+
+/**
+ * @author Daniel Sanz
+ */
+public interface DataSet {
```

Comment:
I like the idea of using perhaps a bit mode concrete name, including one or more of the words "system" and "customizable". Combinations of these with FDS/Dataset lead to:

- `FDS`  (note FDS is the prefix used  in many java classes related to APIs and OSGi components)
- `SystemFDS`
- `CustomizableFDS`
- `CustomizableSystemFDS`
- `DataSet`
- `SystemDataSet`
- `CustomizableDataSet`
- `CustomizableSystemDataSet`

I don't have strong opinion, as long as we keep it clear that this interface must be implemented to expose a system data set to the DSM. That makes me a bit biased towards including "Customizable" AND "System" into the interface name because that exactly captures its purpose